media_cdn/snippets.py

```diff
@@ -149,3 +153,38 @@ def sign_cookie(url_prefix: str, key_name: str, base64_key: str, expiration_time
             policy=policy, signature=signature)
     return signed_policy
 # [END mediacdn_sign_cookie]
+
+
+# [START mediacdn_sign_token]
+def sign_token(url_prefix: str, base64_key: bytes, algo: str, expiration_time: datetime.datetime = None) -> bytes:
+    """Gets the Signed URL Suffix string for the Media CDN' Short token URL requests.
+
+    Args:
+        url_prefix: URL prefix to sign as a string.
+        base64_key: Secret key as a base64 encoded string.
+        algo: Algorithm can be either `SHA1` or `SHA256`.
+        expiration_time: Expiration time as a UTC datetime object.
+
+    Returns:
+        Returns the Signed URL appended with the query parameters based on the
+        specified URL prefix and configuration.
+    """
+    output = b"URLPrefix=" + base64.standard_b64encode(url_prefix.encode("utf-8"))
+
+    if not expiration_time:
+        expiration_time = datetime.datetime.now() + datetime.timedelta(hours=1)
+    epoch_duration = int(
+        (expiration_time - datetime.datetime.utcfromtimestamp(0)).total_seconds()
+    )
+    output += b"~Expires=" + str(epoch_duration).encode("utf-8")
+    key = base64.standard_b64decode(base64_key)
+    algo = algo.lower()
+    if algo == "sha1":
```

Comment:
Should we support SHA1 taking into account it is [no longer considered secure](https://en.wikipedia.org/wiki/SHA-1#:~:text=Since%202005%2C%20SHA%2D1%20has,against%20SHA%2D1%20are%20practical.)? Is Media CDN supporting SHA1?
I would suggest to default to SHA256.

Reply:
Thanks for sharing the wiki link.

In Dual Token, we are using a long & short token and here we are using SHA1 along with a SHA256 secret and HMAC. I believe this is only to support any a legacy customers softwares. But, let me also check with product team if I have any new information.

Reply:
Hi, I have confirmed from the product team.
1. SHA1 is still safe when used along with HMAC
2. SHA1 is supported mainly is for multi-CDN compatibility. Some CDNs only support HMAC-SHA1.

Reply:
SGTM. Thanks for checking.